### ISSUE TITLE
"java.lang.IllegalArgumentException: Illegal character in name 11474" error after upgrading to 2.1.3 from 1.5.2

### ISSUE THREAD

**Issue by ilapavuluri** [dom4j/dom4j#176]
Hi Team,

After upgrading to 2.1.3 from 1.5.2 I am getting java.lang.IllegalArgumentException: Illegal character in name 11474, if my name has number like below:

{"11474" : [{"name" : "dept",
"type" : "9",
"value" : "003|Engineering"},{"name" : "empid",
"type" : "5",
"value" : "spadmin"},{"name" : "empName",
"type" : "5",
"value" : "Administrator, Service Delivery"},{"name" : "reason",
"type" : "12",
"value" : "0"}]}

This was working fine in previous dom4j 1.5.2, if I add a prefix to 11474 like "string_11474", it is working fine.
{"string_11474" : [{"name" : "dept",
"type" : "9",
"value" : "003|Engineering"},{"name" : "empid",
"type" : "5",
"value" : "spadmin"},{"name" : "empName",
"type" : "5",
"value" : "Administrator, Service Delivery"},{"name" : "reason",
"type" : "12",
"value" : "0"}]}

But I am getting this data from external application which I cannot control, how to resolve this issue?
With this I am not able to upgrade from dom4j 1.5.2 to 2.1.3

Appreciate your help.

Thanks
Ashok.

**Comment by ecki** [comment]
Are you creating xml out of those json data? I think xml elements have to start with non-digits. I would add a check in your program to add a prefix. (We use for example _ in some cases).

if you show the whole stack trace it’s easier to tell, but I suspect the error might be from xml parser not from JDom.

**Comment by ecki** [comment]
Well, it looks like a fix to make JDom more complient. This is the change: https://github.com/dom4j/dom4j/commit/c2a99d7dee8ce7a4e5bef134bb781a6672bd8a0f
Not sure if @FilipJirsak would like to make it configurable.
I can only recommend to not rely on not-standard compliant file formats.

**Comment by ilapavuluri** [comment]
yes @ecki I agree with you it is to make it more compliant, but it is another application which I cannot control, **can you make it configurable?** because of this issue I got stuck with old dom4j(1.5.2), with this I am not able to address the vulnerabilities in my product.

Thank you again for your prompt response.
@FilipJirsak can you please make this validation configurable?

Thanks
Ashok.

**Comment by FilipJirsak** [comment]
I don't want to add configuration options to these classes, because testing the configuration options would slow down code for everyone. In the future, I rather consider that it should be possible to use a custom implementation of these classes (i.e. `QName` and `Namespace` would become an interface, as is the case with `Node` and its descendants).
At the moment, probably the only option is to build `dom4j` yourself from source and comment the above test there.
In general, I'd recommend trying to avoid using non-standard XML, because you're bound to run into similar problems in virtually any software where you work with such XML.

**Comment by ilapavuluri** [comment]
Sure Filip, I will try to avoid using non-standard xml itself,

Thank you so much for your response.

Thanks
Ashok.
